app/src/main/java/org/stepic/droid/ui/util/KeyboardHelper.kt

```diff
@@ -20,4 +25,38 @@ inline fun setOnKeyboardOpenListener(
             onKeyboardHidden()
         }
     }
+}
+
+
+fun listenKeyboardChanges(
+        rootView: View,
+        onKeyboardShown: () -> Unit,
+        onKeyboardHidden: () -> Unit): ViewTreeObserver.OnGlobalLayoutListener {
```

Comment:
Looks like duplicate code with method above

Reply:
Unfortunately, method above may not work, when root of "rootView" is not really root and method above will cause memory leak on retain fragments (I will add it to comment)